src/documents/RemoteModelAPI.md

```diff
@@ -0,0 +1,364 @@
+                <dt>payload</dt>
+                <dd>
+                    The data, from <code>that.model.local</code>, to send to the remote source.
+                </dd>
+            </td>
+        </tr>
+    </tbody>
+</table>
+
+## Fetch Workflow
+
+Executing the `fetch` invoker will add a fetch request and returns a promise for the result. Only one request can be in
+flight (processing) at a time. If a write request is in flight, the fetch will be queued. If a fetch request is already
+in queue/flight, the requests will be coalesced into a single fetch request sent to the server, and all representatives
+in the queue will receive the same response payload. When a fetch request is in flight , it will trigger the `fetchImpl`
+invoker to perform the actual request.
+
+Two synthetic events, `onFetch` and `afterFetch`, are fired during the processing of a fetch. `onFetch` can be used to
+perform any necessary actions before running `fetchImpl`. `afterFetch` can be used to perform any necessary actions
+after running `fetchImpl` (e.g. updating the model, unblocking the queue). If promises returned from `onFetch`,
+`afterFetch`, or `fetchImpl` are rejected, the `onFetchError` event will be fired.
+
+After a fetch is performed, the local representation is rebased with the results. The rebasing is performed by
+re-applying the local model changes on top of the values returned from the fetch. In this way the local changes are not
+lost and take precedence over changes pulled down from the remote.
+
+![A flow diagram depicting the Fetch workflow](/images//remoteModel_fetch_diagram.svg)
+
+## Write Workflow
+
+Executing the `write` invoker adds a write request and returns a promise for the result. Only one request can be in
+flight (processing) at a time. If a fetch or write request is in flight, the write will be queued. If a write request is
+already in queue, the requests will be coalesced into a single write request sent to the server, and all representatives
+in the queue will receive the same response payload. When a write request is in flight, it will trigger the `writeImpl`
+invoker to perform the actual request.
+
+Two synthetic events, `onWrite` and `afterWrite`, are fired during the processing of a write. `onWrite` can be used to
+perform any necessary actions before running `writeImpl` (e.g. performing a fetch). `afterWrite` can be used to perform
+any necessary actions after running `writeImpl` (e.g. updating the remote model, unblocking the queue, performing a
+fetch). If promises returned from `onWrite`, `afterWrite`, or `writeImpl` are rejected, the `onWriteError` event will be
+fired.
+
+By default, upon a successful write, the `afterWrite` event will trigger the `remote` model path to be updated with the
+values from the `local` model path. Because of this, the `writeImpl` must resolve/reject its promise after the write
+implementation has been handled by the remote end point, not simply after the request has been made.
```

Comment:
"has been fully handled by the remote endpoint, not simply after the I/O for the request itself has concluded"